src/backend/InvenTree/plugin/builtin/labels/label_sheet.py

```diff
@@ -135,11 +149,15 @@ def print_labels(
         # Render to a single HTML document
         html_data = self.wrap_pages(pages, **document_data)
 
-        # Render HTML to PDF
-        html = weasyprint.HTML(string=html_data)
-        document = html.render().write_pdf()
+        if self.in_debug_mode():
+            # Render HTML to PDF
+            html = weasyprint.HTML(string=html_data)
+            document = html.render().write_pdf()
+
+            output.output = ContentFile(document, 'labels.pdf')
+        else:
+            output.output = ContentFile(html_data, 'labels.html')
```

Comment:
Isn't that just swapped? I mean if in debug mode it should return the html, but you return the pdf.

Reply:
@wolflu05 correct, I was way too hasty on this. @martonmiklos please review

Reply:
Ah, copy paste engineering tricked me, let me fix it.